src/appsignal/config.py

```diff
@@ -1,10 +1,101 @@
+
+DEFAULT_INSTRUMENTATION = Literal[
+    "opentelemetry.instrumentation.celery",
+    "opentelemetry.instrumentation.django",
+    "opentelemetry.instrumentation.jinja2",
+    "opentelemetry.instrumentation.psycopg2",
+    "opentelemetry.instrumentation.redis",
+    "opentelemetry.instrumentation.requests",
+]
+
+DEFAULT_INSTRUMENTATIONS = cast(
+    list[DEFAULT_INSTRUMENTATION], list(get_args(DEFAULT_INSTRUMENTATION))
+)
+
+CONSTANT_PRIVATE_ENVIRON = {"_APPSIGNAL_ENABLE_OPENTELEMETRY_HTTP": "true"}
+
+CONSTANT_RESOURCE_ATTRIBUTES = {
+    "appsignal.config.language_integration": "python",
+}
+
+
+def parse_disable_default_instrumentations(
+    value: Optional[str],
+) -> Optional[Union[list[DEFAULT_INSTRUMENTATION], bool]]:
+    if value is None:
+        return None
+
+    if value == "true":
+        return True
+    if value == "false":
+        return False
+
+    return cast(
+        list[DEFAULT_INSTRUMENTATION],
+        [x for x in value.split(",") if x in DEFAULT_INSTRUMENTATIONS],
+    )
 
 
 class Options(TypedDict, total=False):
     name: str
     environment: str
     push_api_key: str
     log_level: str
+    revision: str
+    disable_default_instrumentations: Union[list[DEFAULT_INSTRUMENTATION], bool]
+
+
+def from_public_environ() -> Options:
+    disable_default_instrumentations = parse_disable_default_instrumentations(
+        os.environ.get("APPSIGNAL_DISABLE_DEFAULT_INSTRUMENTATIONS")
+    )
+
+    # We want the type system to ensure that the keys used here are the
+    # ones defined in the Options typed dict, but we can't directly use
+    # `os.environ.get("APPSIGNAL_WHATEVER")` as their value, as that's an
+    # `Optional[str]`, not an `str`. So we use empty strings as the default
+    # value, then remove the ones whose values are empty strings.
+
+    config = Options(
+        name=os.environ.get("APPSIGNAL_APP_NAME", ""),
+        environment=os.environ.get("APPSIGNAL_APP_ENV", ""),
+        push_api_key=os.environ.get("APPSIGNAL_PUSH_API_KEY", ""),
+        log_level=os.environ.get("APPSIGNAL_LOG_LEVEL", ""),
+        revision=os.environ.get("APP_REVISION", ""),
+    )
+
+    if disable_default_instrumentations is not None:
+        config["disable_default_instrumentations"] = disable_default_instrumentations
+
+    for key, value in list(config.items()):
+        if isinstance(value, str) and value == "":
+            del cast(dict, config)[key]
+
+    return config
+
+
+def set_private_environ(config: Options):
+    private_environ = {
+        "_APPSIGNAL_APP_NAME": config.get("name"),
+        "_APPSIGNAL_PUSH_API_KEY": config.get("push_api_key"),
+        "_APPSIGNAL_ENVIRONMENT": config.get("environment"),
+        "_APPSIGNAL_LOG_LEVEL": config.get("log_level"),
+    } | CONSTANT_PRIVATE_ENVIRON
+
+    for var, value in private_environ.items():
+        if value is not None:
+            os.environ[var] = str(value)
+
+
+def opentelemetry_resource_attributes(config: Options):
```

Comment:
This should also return the app path resource attribute.

Reply:
Thank you! I tried to figure this out, but couldn't, and then I forgot that it needed to be done, because brain.

A quick Google shows this is probably the way to go about this: https://stackoverflow.com/questions/13699283/how-to-get-the-callers-filename-method-name-in-python

Reply:
Sounds good, but we can't rely on the current working dir? Like we do [in the Ruby gem](https://github.com/appsignal/appsignal-ruby/blob/2fc6ba85be1e0cabc2bb8fb26469ad47d1c60243/lib/appsignal.rb#L101).

Reply:
Ah, yes, that should do. I'll go with that instead. Thank you!